server/modules/arenaImport/service/arenaImport/jobs/recordsImportJob.js

```diff
@@ -70,20 +71,47 @@ export default class RecordsImportJob extends Job {
       }
 
       const recordToStore = this.prepareRecordToStore(record)
-      await RecordManager.insertRecord(this.user, surveyId, recordToStore, true, this.tx)
+
+      await this.insertOrSkipRecord({ record: recordToStore, nodesBatchPersister })
+
+      this.incrementProcessedItems()
+    })
+
+    await nodesBatchPersister.flush()
+  }
+
+  async fetchRecordSummary({ recordUuid }) {
+    const { surveyId } = this.context
+    const { list } = await RecordManager.fetchRecordsSummaryBySurveyId({ surveyId, recordUuid }, this.tx)
+    return list[0]
```

Comment:
Why [0]

Reply:
because we need to fetch only one record (summary) in this case. I don't know if it's better to have a function in RecordManager directly to do that, because it's like a fetchRecordSummaryByUuid